network/proxy.sls

```diff
@@ -0,0 +1,65 @@
+{%- from "network/map.jinja" import proxy with context %}
+{%- if proxy.enabled %}
+
+network_proxy_packages:
+  pkg.installed:
+  - names: {{ proxy.pkgs }}
+
+napalm:
+  pip.installed:
+    - name: {{ proxy.pip_pkgs}}
+    - require:
+      - pkg: python-pip
+
+/etc/salt/proxy:
+  file.managed:
+  - source: salt://network/files/proxy.conf.{{ grains.os_family }}
+  - template: jinja
+  - require:
+    - pip: napalm
+
+/usr/lib/python2.7/dist-packages/salt/grains/napalm.py:
+  file.managed:
+  - source: salt://network/files/grains_napalm.py
+  - template: jinja
+  - require:
+    - pip: napalm
+
+
+/usr/lib/python2.7/dist-packages/salt/proxy/napalm.py:
```

Comment:
You shouldn't touch /usr/lib at all + salt-proxy should be setup by salt formula.
Use support metadata to let salt states know what to do.